core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala

```diff
@@ -175,7 +175,7 @@ private[v1] class OneApplicationAttemptResource extends AbstractApplicationResou
   def getAttempt(): ApplicationAttemptInfo = {
     uiRoot.getApplicationInfo(appId)
       .flatMap { app =>
-        app.attempts.filter(_.attemptId == attemptId).headOption
```

Comment:
hmm, just curious, why this doesn't compile in 2.12.7?

Reply:
Yea, can you share the error message?

Reply:
I don't think it's a compile error, but a warning, which might be new in 2.12. I tried to fix up a bunch of these in https://github.com/apache/spark/commit/cfbdd6a1f5906b848c520d3365cc4034992215d9 for example.

Reply:
@sadhen Since this is a general code fix for `master/branch-2.4/branch-2.3`. Can we update the title without mentioning `compiling for Scala 2.12.7`?
> branch-2.2 is fine, but branch-2.3 should be fixed.